lib/install-panel.coffee

```diff
@@ -63,13 +63,14 @@ class InstallPanel extends View
     @loadFeaturedPackages()
 
   detached: ->
+    @packageManagerSubscription.dispose()
     @unsubscribe()
 
   focus: ->
     @searchEditorView.focus()
 
   handleSearchEvents: ->
-    @subscribe @packageManager, 'package-install-failed', (pack, error) =>
+    @packageManagerSubscription = @packageManager.on 'package-install-failed', (pack, error) =>
       @searchErrors.append(new ErrorView(@packageManager, error))
 
     @subscribe atom.commands.add @searchEditorView.element, 'core:confirm', =>
```

Comment:
You don't need to keep the `Subscriber` mix-in for this. You can use `event-kit` here too:

```coffee
@disposables.add atom.commands.add @searchEditorView.element, 'core:confirm', =>
  @performSearch()
```

You just need to this in the `initialize` method:

```coffee
@disposables = new CompositeDisposable
```

This way, you can get rid of the dependency on `emissary`.